google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -659,6 +678,18 @@ private void appendLoop() {
     log.info("Append thread is done. Stream: " + streamName + " id: " + writerId);
   }
 
+  private void throwIfWaitCallbackTooLong() {
+    Duration milliSinceLastCallback =
+        Duration.between(lastRequestCallbackTriggerTime, Instant.now());
```

Comment:
If a StreamWriter is created and then there is a pause before using it, won't this trigger? (odd scenario, I know). Actually if a user ever pauses for 10 minutes, won't this cause the exception the be thrown the next time they use the StreamWriter?

Reply:
The caller of the throwIfWaitCallbackTooLong is checking on the inflight queue size before triggering this check

Reply:
Alternatively, would it make sense to store a timestamp in the RequestWrapper itself when it is added to the inflightRequestQueue? Since the queue is in order, you then can just check the timestamp at the head of the queue

Reply:
right, but append will add to this queue, right? So if append is called after a long delay, the appendLoop might hit this condition (if the appendLoop runs before the callback comes back).

Reply:
Changed to a storing timestamp directly during creation of request wrapper